avocado/core/runner.py

```diff
@@ -427,39 +427,43 @@ def run_suite(self, test_suite, mux, timeout=0, replay_map=None,
-                if deadline is not None and time.time() > deadline:
-                    summary.add('INTERRUPTED')
-                    if 'methodName' in test_parameters:
-                        del test_parameters['methodName']
-                    test_factory = (test.TimeOutSkipTest, test_parameters)
-                    break_loop = not self.run_test(test_factory, queue,
-                                                   summary)
-                    if break_loop:
-                        break
-                else:
-                    if (replay_map is not None and
-                            replay_map[index] is not None):
-                        test_parameters["methodName"] = "test"
-                        test_factory = (replay_map[index], test_parameters)
-
-                    break_loop = not self.run_test(test_factory, queue,
-                                                   summary, deadline)
-                    if break_loop:
-                        break
-            runtime.CURRENT_TEST = None
-            if break_loop:
-                break
+        try:
+            for test_template in test_suite:
+                test_template[1]['base_logdir'] = self.job.logdir
+                test_template[1]['job'] = self.job
+                break_loop = False
+                for test_factory, variant in mux.itertests(test_template):
+                    index += 1
+                    test_parameters = test_factory[1]
+                    name = test_parameters.get("name")
+                    test_parameters["name"] = test.TestName(index + 1, name,
+                                                            variant,
+                                                            no_digits)
+                    if deadline is not None and time.time() > deadline:
+                        summary.add('INTERRUPTED')
+                        if 'methodName' in test_parameters:
+                            del test_parameters['methodName']
+                        test_factory = (test.TimeOutSkipTest, test_parameters)
+                        break_loop = not self.run_test(test_factory, queue,
+                                                       summary)
+                        if break_loop:
+                            break
+                    else:
+                        if (replay_map is not None and
+                                replay_map[index] is not None):
+                            test_parameters["methodName"] = "test"
+                            test_factory = (replay_map[index], test_parameters)
+
+                        break_loop = not self.run_test(test_factory, queue,
+                                                       summary, deadline)
+                        if break_loop:
+                            break
+                runtime.CURRENT_TEST = None
+                if break_loop:
+                    break
+        except KeyboardInterrupt:
```

Comment:
Would you please also add logging.getLogger("avocado.test").error(...) to log the information about this in the result directory?

(while on it you could actually modify the whole `runner` and use the `getLogger` only once at the beginning, but we can also polish it later)